core/src/test/java/com/linecorp/armeria/client/HttpClientIntegrationTest.java

```diff
@@ -300,6 +300,15 @@ public void onComplete() {
                 }, ctx.eventLoop());
                 return res;
             });
+
+            sb.service("glob:/oneparam/**", ((ctx, req) -> {
+                // The client was able to send a request with an escaped path param. Armeria servers always
+                // decode the path so ctx.path == '/oneparam/foo/bar' here.
+                if (req.headers().path().equals("/oneparam/foo%2Fbar")) {
```

Comment:
How about checking `decodedPath()` as well?

Reply:
Added the check for `ctx.path()` - it does make me wonder, though, why is there a `decodedPath()` if `path()` is already decoded? :thinking: